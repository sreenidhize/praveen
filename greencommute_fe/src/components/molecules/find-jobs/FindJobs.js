import { Divider, Grid, makeStyles } from "@material-ui/core";
import React from "react";
import IconInputField from "../../molecules/icon-input-fiield/IconInputField";
import LocationOnIcon from "@material-ui/icons/LocationOn";
import SearchIcon from "@material-ui/icons/Search";
import Button from "../../atom/button/Button";

const useStyles = makeStyles((theme) => ({
  halfButton: {
    borderTopLeftRadius: 0,
    borderBottomLeftRadius: 0,
    height: "50px",
  },
  itemMargin: {
    backgroundColor: "white",
    borderRadiusRight: "10px",
    borderLeft: "1px solid #9bbdcb ",
  },
  root: {
    width: "100%",
    height: "50px",
  },
  icon: {
    paddingTop: "10px",
    paddingLeft: "4px",
  },
}));

const FindJobs = (props) => {
  const classes = useStyles();
  const { defaultSkillValue, defaultLocationValue } = props;

  return (
    <Grid container direction="row">
      <Grid container item xs={12} className={classes.root}>
        <Grid
          item
          container
          direction="column"
          xs={5}
          style={{
            backgroundColor: "white",
            borderRadiusLeft: "10px",
          }}
        >
          <IconInputField
            placeholder="Search Skills"
            defaultValue={defaultSkillValue}
          />
        </Grid>
        <Grid item container xs={6} className={classes.itemMargin}>
          <IconInputField
            icon={<LocationOnIcon color="secondary" className={classes.icon} />}
            placeholder="Location"
            defaultValue={defaultLocationValue}
          />
        </Grid>
        <Grid item style={{ justify: "flex-end" }}>
          <Button
            variant="contained"
            color="primary"
            type="long"
            className={classes.halfButton}
          >
            <SearchIcon />
          </Button>
        </Grid>
      </Grid>
    </Grid>
  );
};

export default FindJobs;
