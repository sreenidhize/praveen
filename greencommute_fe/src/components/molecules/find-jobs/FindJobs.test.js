import React from "react";
import FindJobs from "./FindJobs";
import { render } from "@testing-library/react";

describe("component renders correctly", () => {
  test("should match snapshot", () => {
    const { asFragment, debug } = render(<FindJobs />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("icon and placeholder should not be empty", () => {
    const { debug, container } = render(<FindJobs />);

    expect(container.getElementsByTagName("div").length).toBe(15);
  });
});
