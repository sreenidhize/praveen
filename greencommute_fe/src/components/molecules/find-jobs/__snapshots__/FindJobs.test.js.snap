// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders correctly should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root makeStyles-root-3 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-5"
        style="background-color: white;"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-6 MuiSvgIcon-colorSecondary"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 6V4h-4v2h4zM4 8v11h16V8H4zm16-2c1.11 0 2 .89 2 2v11c0 1.11-.89 2-2 2H4c-1.11 0-2-.89-2-2l.01-11c0-1.11.88-2 1.99-2h4V4c0-1.11.89-2 2-2h4c1.11 0 2 .89 2 2v2h4z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-text-5"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInputBase-colorSecondary MuiInput-colorSecondary MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  placeholder="Search Skills"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-itemMargin-2 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-4 MuiSvgIcon-colorSecondary"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
              />
            </svg>
          </div>
          <div
            class="MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-text-5"
            >
              <div
                class="MuiInputBase-root MuiInput-root MuiInputBase-colorSecondary MuiInput-colorSecondary MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  placeholder="Location"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-halfButton-1 MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
